styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/PublisherHandler.java

```diff
@@ -40,16 +46,28 @@
   private static final Logger LOG = LoggerFactory.getLogger(PublisherHandler.class);
 
   private static final int MAX_RETRIES = 420;
-  private static final Retrier RETRIER = Retrier.builder()
-      .errorMessage("publish deploy event")
-      .retryDelay(Duration.ofSeconds(1))
-      .maxRetries(MAX_RETRIES)
-      .build();
 
+  private static final String DEPLOYING = "deploying";
+  private static final String DEPLOYED = "deployed";
+
+  private final Retrier retrier;
   private final Publisher publisher;
+  private final Stats stats;
 
-  public PublisherHandler(Publisher publisher) {
+  public PublisherHandler(Publisher publisher, Stats stats) {
+    this(publisher, stats,
+        Retrier.builder()
+            .errorMessage("publish deploy event")
+            .retryDelay(Duration.ofSeconds(1))
```

Comment:
Exponential backoff?

Reply:
Could be, but not in this PR I think.

Reply:
If we really think it is important to send this event, then we should persist it to durable storage.

Reply:
I don't actually know the initial requirement of ITGC.

Reply:
This has been very reliable, meaning ~100% in the past years. Not sure it is really worth persisting that includes more moving parts.